src/gt4py/backend/pyext_builder.py

```diff
@@ -90,6 +93,14 @@ def get_gt_pyext_build_opts(
             extra_link_args=extra_link_args,
         )
 
+    if uses_openmp:
```

Comment:
In general I fully agree with factoring common code as you do here, but in this case I think it's a bit overkill and makes the code less readable instead. What would you think about removing the loop and do something simpler? 

Example:
```
if gt_config.build_settings["openmp_cppflags"]:
    build_opts["extra_compile_args"].append(gt_config.build_settings["openmp_cppflags")
if gt_config.build_settings["openmp_ldflags"]:
    build_opts["extra_link_args"].append(gt_config.build_settings["openmp_ldflags")
```

Reply:
I agree, that zip is gnarly. I've simplified it now. Let me know if you still would like changes made!